packages/ra-auth-google/Readme.md

```diff
@@ -0,0 +1,581 @@
+- Using the Google Identity Service JavaScript library [with the token model](https://developers.google.com/identity/oauth2/web/guides/use-token-model) to call a Google API
+- Using the Google Identity Service JavaScript library [with the token model](https://developers.google.com/identity/oauth2/web/guides/use-token-model), along with the [Google API Client Library for JavaScript](https://github.com/google/google-api-javascript-client/blob/master/docs/samples.md#loading-an-api-and-making-a-request), to call a Google API
+- Using the [Admin SDK API](https://developers.google.com/admin-sdk/reference-overview?hl=en) to retrieve all groups/roles of a user, to compute permissions
+
+## Limitations
+
+Since they would require a back-end, the following features are not supported by this package:
+
+- Automatic [token refresh](https://developers.google.com/identity/oauth2/web/guides/use-code-model?hl=en#get_access_and_refresh_tokens)
+- Sign in with Google button in [redirect mode](https://developers.google.com/identity/gsi/web/reference/js-reference?hl=en#login_uri)
+
+It's up to you to implement these features in your back-end if you need them.
+
+## Installation
+
+```sh
+yarn add ra-auth-google
+# or
+npm install --save ra-auth-google
+```
+
+## Basic Usage
+
+The following example implements [Sign in with Google button](https://developers.google.com/identity/gsi/web/guides/offerings?hl=en#sign_in_with_google_button) in popup mode, using the client id configured in the `.env` file.
+
+```sh
+# in .env
+VITE_GOOGLE_CLIENT_ID="my-application-client-id.apps.googleusercontent.com"
+```
+
+```html
+<!-- in index.html -->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <!-- Add the following line to load the Google Identity Services library -->
+    <script async defer src="https://accounts.google.com/gsi/client"></script>
+  </head>
+  <body>
+    <!-- ... -->
+  </body>
+</html>
+```
+
+```tsx
+// in src/App.tsx
+import React from "react";
+import { Admin, Resource, Login } from "react-admin";
+import { initGoogleAuthProvider } from "ra-auth-google";
+import dataProvider from "./dataProvider";
+import posts from "./posts";
+
+const App = () => {
+  const { authProvider, LoginButton } = initGoogleAuthProvider();
+
+  const LoginPage = () => (
+    <Login>
+      <LoginButton theme="filled_black" />
+    </Login>
+  );
+
+  return (
+    <Admin
+      authProvider={authProvider}
+      dataProvider={dataProvider}
+      title="Example Admin"
+      loginPage={LoginPage}
+    >
+      <Resource name="posts" {...posts} />
+    </Admin>
+  );
+};
+export default App;
+```
+
+**Tip:** This example uses the [`<Login>`](https://marmelab.com/react-admin/Authentication.html#customizing-the-login-component) component provided by react-admin to create the login page. You can also include the `<LoginButton>` component in your own login page if you prefer.
+
+![LoginPage](img/LoginPage.png)
+
+## Advanced Usage
```

Comment:
I'd remove this section title, and move all the child sections up one level (## instead of ###)